### ISSUE TITLE
Program Exception - illegal instruction

### ISSUE THREAD

**Issue by miroi** [firemodels/fds#10989]
Hello,

I downloaded and installed fresh fds 6.7.9 and installed it on our (virtual) Linux x86_64 Debian school server.

Can not run the executable fsb due to error - see below.

~~~
milias@labs.fpv.umb.sk:~/work/open-collection/firemodels/fds/runs/sample/.bash_run-labs.01


 Running  FDS ...
sourcing /home/milias/FDS/FDS6/bin/FDS6VARS.sh

FI_PROVIDER_PATH=/home/milias/FDS/FDS6/bin/INTEL/prov
LD_LIBRARY_PATH=/home/milias/FDS/FDS6/bin/INTEL/lib:/usr/lib64:
PATH=/home/milias/FDS/FDS6/bin/INTEL/bin:/home/milias/FDS/FDS6/bin:/home/milias/.local/bin:/home/milias/.local/bin:/home/milias/work/software/cmake/cmake-3.22.2-linux-x86_64/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:.
OMP_NUM_THREADS=4
position of fds : /home/milias/FDS/FDS6/bin/fds
linked libraries, ldd fds:
        linux-vdso.so.1 (0x00007ffc1536b000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f9451d06000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9451b83000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f9451b7e000)
        libmpifort.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpifort.so.12 (0x00007f94517ca000)
        libmpi.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpi.so.12 (0x00007f944ff82000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f944ff76000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f944fdb5000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f9451d4f000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f944fd9b000)
/home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec
Intel(R) MPI Library for Linux* OS, Version 2021.6 Build 20220227 (id: 28877f3f32)
Copyright 2003-2022, Intel Corporation.

 running mpiexec -n 2 /home/milias/FDS/FDS6/bin/fds :

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  13:40:21
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  13:40:21
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 running mpiexec -n 2 /home/milias/FDS/FDS6/bin/fds sample.fds :
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F0FC6837730  Unknown               Unknown  Unknown
libmpi.so.12       00007F0FC564DEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F0FC564907A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F0FC4F366CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F0FC63AC880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F0FC48F809B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F8861238730  Unknown               Unknown  Unknown
libmpi.so.12       00007F886004EEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F886004A07A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F885F9376CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F8860DAD880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F885F2F909B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
milias@labs.fpv.umb.sk:~/work/open-collection/firemodels/fds/runs/sample/.
~~~

we have:

~~~
milias@labs.fpv.umb.sk:~/FDS/FDS6/Examples/Fires/.cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
milias@labs.fpv.umb.sk:~/FDS/FDS6/Examples/Fires/.uname -a
Linux labs.fpv.umb.sk 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux
~~~

**Comment by marcosvanella** [comment]
For some reason a two mpi process job requested is seen by your system as two one mpi process runs. You might have a mixup of different mpi libs. Type "which mpiexec" and report back.On Oct 8, 2022 7:46 AM, Miro Iliaš ***@***.***> wrote:
Hello,
I downloaded and installed fresh fds 6.7.9 and installed it on our (virtual) Linux x86_64 Debian school server.
Can not run the executable fsb due to error - see below.
***@***.***:~/work/open-collection/firemodels/fds/runs/sample/.bash_run-labs.01


 Running  FDS ...
sourcing /home/milias/FDS/FDS6/bin/FDS6VARS.sh

FI_PROVIDER_PATH=/home/milias/FDS/FDS6/bin/INTEL/prov
LD_LIBRARY_PATH=/home/milias/FDS/FDS6/bin/INTEL/lib:/usr/lib64:
PATH=/home/milias/FDS/FDS6/bin/INTEL/bin:/home/milias/FDS/FDS6/bin:/home/milias/.local/bin:/home/milias/.local/bin:/home/milias/work/software/cmake/cmake-3.22.2-linux-x86_64/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:.
OMP_NUM_THREADS=4
position of fds : /home/milias/FDS/FDS6/bin/fds
linked libraries, ldd fds:
        linux-vdso.so.1 (0x00007ffc1536b000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f9451d06000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9451b83000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f9451b7e000)
        libmpifort.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpifort.so.12 (0x00007f94517ca000)
        libmpi.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpi.so.12 (0x00007f944ff82000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f944ff76000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f944fdb5000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f9451d4f000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f944fd9b000)
/home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec
Intel(R) MPI Library for Linux* OS, Version 2021.6 Build 20220227 (id: 28877f3f32)
Copyright 2003-2022, Intel Corporation.

 running mpiexec -n 2 /home/milias/FDS/FDS6/bin/fds :

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  13:40:21
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  13:40:21
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 running mpiexec -n 2 /home/milias/FDS/FDS6/bin/fds sample.fds :
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F0FC6837730  Unknown               Unknown  Unknown
libmpi.so.12       00007F0FC564DEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F0FC564907A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F0FC4F366CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F0FC63AC880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F0FC48F809B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F8861238730  Unknown               Unknown  Unknown
libmpi.so.12       00007F886004EEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F886004A07A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F885F9376CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F8860DAD880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F885F2F909B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
***@***.***:~/work/open-collection/firemodels/fds/runs/sample/.

we have:
***@***.***:~/FDS/FDS6/Examples/Fires/.cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
***@***.***:~/FDS/FDS6/Examples/Fires/.uname -a
Linux labs.fpv.umb.sk 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

On Oct 8, 2022 7:46 AM, Miro Iliaš ***@***.***> wrote:
Hello,
I downloaded and installed fresh fds 6.7.9 and installed it on our (virtual) Linux x86_64 Debian school server.
Can not run the executable fsb due to error - see below.
***@***.***:~/work/open-collection/firemodels/fds/runs/sample/.bash_run-labs.01


 Running  FDS ...
sourcing /home/milias/FDS/FDS6/bin/FDS6VARS.sh

FI_PROVIDER_PATH=/home/milias/FDS/FDS6/bin/INTEL/prov
LD_LIBRARY_PATH=/home/milias/FDS/FDS6/bin/INTEL/lib:/usr/lib64:
PATH=/home/milias/FDS/FDS6/bin/INTEL/bin:/home/milias/FDS/FDS6/bin:/home/milias/.local/bin:/home/milias/.local/bin:/home/milias/work/software/cmake/cmake-3.22.2-linux-x86_64/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:.
OMP_NUM_THREADS=4
position of fds : /home/milias/FDS/FDS6/bin/fds
linked libraries, ldd fds:
        linux-vdso.so.1 (0x00007ffc1536b000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f9451d06000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9451b83000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f9451b7e000)
        libmpifort.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpifort.so.12 (0x00007f94517ca000)
        libmpi.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpi.so.12 (0x00007f944ff82000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f944ff76000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f944fdb5000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f9451d4f000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f944fd9b000)
/home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec
Intel(R) MPI Library for Linux* OS, Version 2021.6 Build 20220227 (id: 28877f3f32)
Copyright 2003-2022, Intel Corporation.

 running mpiexec -n 2 /home/milias/FDS/FDS6/bin/fds :

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  13:40:21
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  13:40:21
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 running mpiexec -n 2 /home/milias/FDS/FDS6/bin/fds sample.fds :
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F0FC6837730  Unknown               Unknown  Unknown
libmpi.so.12       00007F0FC564DEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F0FC564907A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F0FC4F366CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F0FC63AC880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F0FC48F809B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F8861238730  Unknown               Unknown  Unknown
libmpi.so.12       00007F886004EEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F886004A07A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F885F9376CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F8860DAD880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F885F2F909B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
***@***.***:~/work/open-collection/firemodels/fds/runs/sample/.

we have:
***@***.***:~/FDS/FDS6/Examples/Fires/.cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
***@***.***:~/FDS/FDS6/Examples/Fires/.uname -a
Linux labs.fpv.umb.sk 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

On Oct 8, 2022 7:46 AM, Miro Iliaš ***@***.***> wrote:
Hello,
I downloaded and installed fresh fds 6.7.9 and installed it on our (virtual) Linux x86_64 Debian school server.
Can not run the executable fsb due to error - see below.
***@***.***:~/work/open-collection/firemodels/fds/runs/sample/.bash_run-labs.01


 Running  FDS ...
sourcing /home/milias/FDS/FDS6/bin/FDS6VARS.sh

FI_PROVIDER_PATH=/home/milias/FDS/FDS6/bin/INTEL/prov
LD_LIBRARY_PATH=/home/milias/FDS/FDS6/bin/INTEL/lib:/usr/lib64:
PATH=/home/milias/FDS/FDS6/bin/INTEL/bin:/home/milias/FDS/FDS6/bin:/home/milias/.local/bin:/home/milias/.local/bin:/home/milias/work/software/cmake/cmake-3.22.2-linux-x86_64/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:.
OMP_NUM_THREADS=4
position of fds : /home/milias/FDS/FDS6/bin/fds
linked libraries, ldd fds:
        linux-vdso.so.1 (0x00007ffc1536b000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f9451d06000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9451b83000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f9451b7e000)
        libmpifort.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpifort.so.12 (0x00007f94517ca000)
        libmpi.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpi.so.12 (0x00007f944ff82000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f944ff76000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f944fdb5000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f9451d4f000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f944fd9b000)
/home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec
Intel(R) MPI Library for Linux* OS, Version 2021.6 Build 20220227 (id: 28877f3f32)
Copyright 2003-2022, Intel Corporation.

 running mpiexec -n 2 /home/milias/FDS/FDS6/bin/fds :

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  13:40:21
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  13:40:21
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 running mpiexec -n 2 /home/milias/FDS/FDS6/bin/fds sample.fds :
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F0FC6837730  Unknown               Unknown  Unknown
libmpi.so.12       00007F0FC564DEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F0FC564907A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F0FC4F366CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F0FC63AC880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F0FC48F809B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F8861238730  Unknown               Unknown  Unknown
libmpi.so.12       00007F886004EEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F886004A07A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F885F9376CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F8860DAD880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F885F2F909B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
***@***.***:~/work/open-collection/firemodels/fds/runs/sample/.

we have:
***@***.***:~/FDS/FDS6/Examples/Fires/.cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
***@***.***:~/FDS/FDS6/Examples/Fires/.uname -a
Linux labs.fpv.umb.sk 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux

**Comment by miroi** [comment]
Hello, thanks for your reply.

I have  /home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec installed from FDS, which I am using:
~~~
 ldd /home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec :
        linux-vdso.so.1 (0x00007ffc911cb000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f46ae2da000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f46ae2b9000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f46ae136000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f46ae11c000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f46adf5b000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f46ae307000)
~~~

also I have default installation at that machine:
~~~
milias@labs.fpv.umb.sk:~/work/open-collection/firemodels/fds/runs/sample/.which mpiexec
/usr/bin/mpiexec
milias@labs.fpv.umb.sk:~/work/open-collection/firemodels/fds/runs/sample/.ldd /usr/bin/mpiexec
        linux-vdso.so.1 (0x00007fff18b9a000)
        libopen-rte.so.40 => /lib/x86_64-linux-gnu/libopen-rte.so.40 (0x00007f45e04c3000)
        libopen-pal.so.40 => /lib/x86_64-linux-gnu/libopen-pal.so.40 (0x00007f45e0416000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f45e040c000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f45e0289000)
        libutil.so.1 => /lib/x86_64-linux-gnu/libutil.so.1 (0x00007f45e0284000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f45e0066000)
        libhwloc.so.5 => /lib/x86_64-linux-gnu/libhwloc.so.5 (0x00007f45e0022000)
        libevent-2.1.so.6 => /lib/x86_64-linux-gnu/libevent-2.1.so.6 (0x00007f45dfdcc000)
        libevent_pthreads-2.1.so.6 => /lib/x86_64-linux-gnu/libevent_pthreads-2.1.so.6 (0x00007f45dfbc9000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f45dfba8000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f45df9e7000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f45df9e2000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f45e05a8000)
        libnuma.so.1 => /lib/x86_64-linux-gnu/libnuma.so.1 (0x00007f45df9d2000)
        libltdl.so.7 => /lib/x86_64-linux-gnu/libltdl.so.7 (0x00007f45df9c7000)
~~~

Though I have proper mpiexec, still have 2 MPI processes, instead of one with 2 threads:
~~~
 running /home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec -n 2 /home/milias/FDS/FDS6/bin/fds :

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  16:39:46
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 Fire Dynamics Simulator

 Current Date     : October  8, 2022  16:39:46
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Disabled

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS

 Consult FDS Users Guide Chapter, Running FDS, for further instructions.
~~~

How to fix that ?

**Comment by miroi** [comment]
I see "test_mpi" is installed within FDS, now I see missing libraries...this could be the reason for failures mentioned above.

~~~
 which test_mpi : /home/milias/FDS/FDS6/bin/test_mpi
ldd /home/milias/FDS/FDS6/bin/test_mpi :        linux-vdso.so.1 (0x00007ffdcd1f9000)
        libVTmc.so => not found
        libdwarf.so.1 => not found
        libelf.so.1 => /lib/x86_64-linux-gnu/libelf.so.1 (0x00007f217ffce000)
        libvtunwind.so => not found
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f217fe4b000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f217fe28000)
        libmpifort.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpifort.so.12 (0x00007f217fa74000)
        libmpi.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpi.so.12 (0x00007f217e22c000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f217e227000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f217e21d000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f217e05c000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f2180012000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f217e040000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f217de22000)
~~~

**Comment by marcosvanella** [comment]
Have you ever run an mpi executable in parallel (not FDS) in this machine correctly? I remember in the past we had instances in our cluster where this would happen (mpirun launching serial versions of the job in the np mpi processes requested). I can't remember the actual reason. Maybe Glenn or Kevin can.On Oct 8, 2022 11:10 AM, Miro Iliaš ***@***.***> wrote:
I see "test_mpi" is installed within FDS, now I see missing libraries...this could be the reason for failures mentioned above.
 which test_mpi : /home/milias/FDS/FDS6/bin/test_mpi
ldd /home/milias/FDS/FDS6/bin/test_mpi :        linux-vdso.so.1 (0x00007ffdcd1f9000)
        libVTmc.so => not found
        libdwarf.so.1 => not found
        libelf.so.1 => /lib/x86_64-linux-gnu/libelf.so.1 (0x00007f217ffce000)
        libvtunwind.so => not found
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f217fe4b000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f217fe28000)
        libmpifort.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpifort.so.12 (0x00007f217fa74000)
        libmpi.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpi.so.12 (0x00007f217e22c000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f217e227000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f217e21d000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f217e05c000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f2180012000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f217e040000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f217de22000)

**Comment by miroi** [comment]
Yes, I am running MPI program on this machine - but only with GNU OpenMPI, no IntelMPI.   In between,  I encountered the same problem at other machine - our university cluster - where I am missing the same libraries. After  installing missing libraries I will return to this issue.

**Comment by miroi** [comment]
Hello,

could you please tell us what packages on Debian are for missing libraries ?

       libVTmc.so => not found
        libdwarf.so.1 => not found
        libvtunwind.so => not found

And also, by which launching command can we avoid mixing of brought-in intel mpi with native installed openmpi libs ?

**Comment by marcosvanella** [comment]
I got a virtual machine with Ubuntu 14 (debian based) and installed fds6.7.9. I see the same libraries missing from test_mpi but they are not required by fds. This is what I get for linked libraries:

```
~/FDS/FDS6/bin$ ldd fds
        linux-vdso.so.1 =>  (0x00007ffe20be0000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00002ad2ffcc3000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00002ad2ffee1000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00002ad3001e7000)
        libmpifort.so.12 => /home/mnv/FDS/FDS6/bin/INTEL/lib/libmpifort.so.12 (0x00002ad3003eb000)
        libmpi.so.12 => /home/mnv/FDS/FDS6/bin/INTEL/lib/libmpi.so.12 (0x00002ad30079f000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00002ad301fe7000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00002ad3021ef000)
        /lib64/ld-linux-x86-64.so.2 (0x00002ad2ffa9e000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00002ad3025b8000)
```

VT refers to the Intel Trace collector and comes with the Intel OneAPI packages. I noted that trying to run test_mpi the program crashes with a missing libraries error.
I don't think the problem you see is related to these missing libraries, as I can run fds correctly in parallel in the virtual machine.
As long as you are pointing to the right mpiexec (Intel), in theory, you should not have issues of mixing mpi libraries.
Something you can do to check this, if you have admin rights in your computer, is to uninstall openmpi and try running fds with intel mpi.
On my end I'm downloading debian GNU Linux 9 on my Parallels VM and will see next if I can reproduce this.

**Comment by marcosvanella** [comment]
No need to erase the openmpi version you have. I note in both computers that typing mpiexec -n 2 fds launches two serial jobs of fds that type the same header. So this looks like a red herring.
Go to ~/FDS/FDS6/Examples/Pressure_Solver and type : mpiexec -n 4 dancing_eddies_default.fds

This is what I get on my end with Debian Gnu Linux 9:

parallels@debian-gnu-linux-vm:~/FDS/FDS6/Examples/Pressure_Solver$ mpiexec -n 4 fds dancing_eddies_default.fds 

 Starting FDS ...

 MPI Process      0 started on debian-gnu-linux-vm
 MPI Process      3 started on debian-gnu-linux-vm
 MPI Process      2 started on debian-gnu-linux-vm
 MPI Process      1 started on debian-gnu-linux-vm

 Reading FDS input file ...


 Fire Dynamics Simulator

 Current Date     : October 11, 2022  11:19:50
 Revision         : FDS6.7.9-0-gec52dee42-release
 Revision Date    : Sun Jun 26 14:36:40 2022 -0400
 Compiler         : ifort version 2021.6.0
 Compilation Date : Jun 28, 2022 23:02:23

 MPI Enabled;    Number of MPI Processes:       4
 OpenMP Disabled

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2021.6 for Linux* OS


 Job TITLE        : Simple 2D Tunnel
 Job ID string    : dancing_eddies_default

 Time Step:      1, Simulation Time:     0.004 s
 Time Step:      2, Simulation Time:     0.008 s
 Time Step:      3, Simulation Time:      0.01 s
 Time Step:      4, Simulation Time:      0.02 s
 Time Step:      5, Simulation Time:      0.02 s
 Time Step:      6, Simulation Time:      0.02 s
 Time Step:      7, Simulation Time:      0.03 s

**Comment by miroi** [comment]
Well, without test_mpi, still getting the error:
~~~
milias@labs.fpv.umb.sk:~/work/open-collection/firemodels/fds/runs/Pressure_Solver/.ls
bash_run-labs.01*  dancing_eddies_default.fds  readme.rst
milias@labs.fpv.umb.sk:~/work/open-collection/firemodels/fds/runs/Pressure_Solver/.bash_run-labs.01

 Running  FDS ...
sourcing /home/milias/FDS/FDS6/bin/FDS6VARS.sh

FI_PROVIDER_PATH=/home/milias/FDS/FDS6/bin/INTEL/prov
LD_LIBRARY_PATH=/home/milias/FDS/FDS6/bin/INTEL/lib:/usr/lib64:
PATH=/home/milias/FDS/FDS6/bin/INTEL/bin:/home/milias/FDS/FDS6/bin:/home/milias/.local/bin:/home/milias/.local/bin:/home/milias/work/software/cmake/cmake-3.22.2-linux-x86_64/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:.
OMP_NUM_THREADS=4
position of fds : /home/milias/FDS/FDS6/bin/fds
linked libraries, ldd fds:
        linux-vdso.so.1 (0x00007ffe554da000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f6bd91fa000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f6bd9077000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f6bd9072000)
        libmpifort.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpifort.so.12 (0x00007f6bd8cbe000)
        libmpi.so.12 => /home/milias/FDS/FDS6/bin/INTEL/lib/libmpi.so.12 (0x00007f6bd7476000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f6bd746a000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f6bd72a9000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f6bd9244000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f6bd728f000)
MPIEXEC=/home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec
Intel(R) MPI Library for Linux* OS, Version 2021.6 Build 20220227 (id: 28877f3f32)
Copyright 2003-2022, Intel Corporation.

 ldd /home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec :
        linux-vdso.so.1 (0x00007fffb12c6000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f9c3ddc8000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f9c3dda7000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9c3dc24000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f9c3dc0a000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f9c3da49000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f9c3ddf6000)

 running /home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec -n 4 /home/milias/FDS/FDS6/bin/fds dancing_eddies_default.fds :
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F04E6AC1730  Unknown               Unknown  Unknown
libmpi.so.12       00007F04E58D7EDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F04E58D307A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F04E51C06CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F04E6636880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F04E4B8209B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F2279C78730  Unknown               Unknown  Unknown
libmpi.so.12       00007F2278A8EEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F2278A8A07A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F22783776CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F22797ED880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F2277D3909B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F5A0FB25730  Unknown               Unknown  Unknown
libmpi.so.12       00007F5A0E93BEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F5A0E93707A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F5A0E2246CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F5A0F69A880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F5A0DBE609B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007F7B757EA730  Unknown               Unknown  Unknown
libmpi.so.12       00007F7B74600EDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007F7B745FC07A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007F7B73EE96CF  PMPI_Init_thread      Unknown  Unknown
libmpifort.so.12   00007F7B7535F880  mpi_init_thread_f     Unknown  Unknown
fds                0000000006AFEE91  Unknown               Unknown  Unknown
fds                0000000000407112  Unknown               Unknown  Unknown
libc-2.28.so       00007F7B738AB09B  __libc_start_main     Unknown  Unknown
fds                0000000000407029  Unknown               Unknown  Unknown
milias@labs.fpv.umb.sk:~/work/open-collection/firemodels/fds/runs/Pressure_Solver/.
~~~

**Comment by marcosvanella** [comment]
Is this a cluster that you use .bash_run-labs.01 to submit your calculation? Can you (just to test) type 

/home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec -n 4 /home/milias/FDS/FDS6/bin/fds dancing_eddies_default.fds

in the head node prompt?

I cannot reproduce this issue.

**Comment by miroi** [comment]
No, labs is a debian buster virtual linux server, for our students mainly. I prefer bash-file script to commulate usefull commands inside. 
~~~
milias@labs.fpv.umb.sk:~/work/open-collection/firemodels/fds/runs/Pressure_Solver/.source /home/milias/FDS/FDS6/bin/FDS6VARS.sh     
milias@labs.fpv.umb.sk:~/work/open-collection/firemodels/fds/runs/Pressure_Solver/./home/milias/FDS/FDS6/bin/INTEL/bin/mpiexec -n 4 /home/milias/FDS/FDS6/bin/fds  dancing_eddies_default.fds
forrtl: severe (168): Program Exception - illegal instruction
Image              PC                Routine            Line        Source
fds                0000000006A3286B  Unknown               Unknown  Unknown
libpthread-2.28.s  00007FF0242B5730  Unknown               Unknown  Unknown
libmpi.so.12       00007FF0230CBEDF  MPL_wtime_init        Unknown  Unknown
libmpi.so.12       00007FF0230C707A  MPL_dbg_pre_init      Unknown  Unknown
libmpi.so.12       00007FF0229B46CF  PMPI_Init_thread      Unknown  Unknown
.
.
~~~

**Comment by mcgratta** [comment]
I think we have run out of suggestions. One last thing -- compile the following simple Fortran program and run it on your cluster. This program has nothing to do with FDS. It is just a simple Fortran code.
```
        program test_mpi
        use mpi
        implicit none

        integer i, size, rank, namelen, ierr
        character (len=MPI_MAX_PROCESSOR_NAME) :: name
        integer stat(MPI_STATUS_SIZE)

        call MPI_INIT (ierr)

        call MPI_COMM_SIZE (MPI_COMM_WORLD, size, ierr)
        call MPI_COMM_RANK (MPI_COMM_WORLD, rank, ierr)
        call MPI_GET_PROCESSOR_NAME (name, namelen, ierr)

        if (rank.eq.0) then

            print *, 'Hello world: rank ', rank, ' of ', size, ' running on ', name

            do i = 1, size - 1
                call MPI_RECV (rank, 1, MPI_INTEGER, i, 1, MPI_COMM_WORLD, stat, ierr)
                call MPI_RECV (size, 1, MPI_INTEGER, i, 1, MPI_COMM_WORLD, stat, ierr)
                call MPI_RECV (namelen, 1, MPI_INTEGER, i, 1, MPI_COMM_WORLD, stat, ierr)
                name = ''
                call MPI_RECV (name, namelen, MPI_CHARACTER, i, 1, MPI_COMM_WORLD, stat, ierr)
                print *, 'Hello world: rank ', rank, ' of ', size, ' running on ', name
            enddo

        else

            call MPI_SEND (rank, 1, MPI_INTEGER, 0, 1, MPI_COMM_WORLD, ierr)
            call MPI_SEND (size, 1, MPI_INTEGER, 0, 1, MPI_COMM_WORLD, ierr)
            call MPI_SEND (namelen, 1, MPI_INTEGER, 0, 1, MPI_COMM_WORLD, ierr)
            call MPI_SEND (name, namelen, MPI_CHARACTER, 0, 1, MPI_COMM_WORLD, ierr)

        endif

        call MPI_FINALIZE (ierr)

        end
```

**Comment by marcosvanella** [comment]
My take is that you will need to compile FDS for your system. The fact that I cannot reproduce the issue on my Virtual Machine with both Ubuntu and Debian GNU Linux, + the illegal instruction error you get, tells me that the executable shipped in the FDS bundle probably has optimizations that your CPU does not handle. Some context here on a particular cluster:
https://rc.byu.edu/wiki/index.php?page=I+keep+getting+Illegal+Instruction+errors.+What+do+I+do%3F

**Comment by miroi** [comment]
Hello, the short MPI program works, see https://github.com/miroi/short_programs/blob/main/test_mpi/readme.rst . I agree that I have to compile the code from the scratch, using native GNU OpenMPI compilers.

**Comment by mcgratta** [comment]
There are some notes [here](https://github.com/firemodels/fds/wiki/Compiling-FDS-with-GNU-Fortran-in-Ubuntu-Linux).

**Comment by dl1ycf** [comment]
I just came accross through a search engine. I think the problem is Intel MPI related. When Intel 2019 came out, I got the same problem one one cluster (on the other it worked). It could never be sorted out, so I switched from Intel MPI to
OpenMPI few years back, and everything is fine. It is perfectly possible to use OpenMPI with ifort. The only thing you have to know is that OpenMPI cannot do 32-bit and 64-bit FORTRAN integers BOTH, you need to compile the OpenMPI library twice (with 32 and 64 bit fortran ints) and take care to use the correct one in your project. This was more easy with  IntelMPI.

**Comment by marcosvanella** [comment]
Closing this. Open another issue if you have compilation problems.
